### PR TITLE
improve vpn_customer_gateway testcase

### PR DESCRIPTION
**Pull request by ddaddaddda**

### CODE REVIEW

File: alicloud/service_alicloud_vpn_gateway.go [hunk 1]
```diff
@@ -37,24 +37,25 @@ func (s *VpnGatewayService) DescribeVpnGateway(id string) (v vpc.DescribeVpnGate
...
-	request.CustomerGatewayId = cgwId
+	request.CustomerGatewayId = id
 
 	raw, err := s.client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
 		return vpcClient.DescribeCustomerGateway(request)
 	})
 	if err != nil {
 		if IsExceptedErrors(err, []string{VpnForbidden, CgwNotFound}) {
-			return v, GetNotFoundErrorFromString(GetNotFoundMessage("VPN customer gateway", cgwId))
+			return v, WrapErrorf(Error(GetNotFoundMessage("VpnCustomerGateway", id)), NotFoundMsg, ProviderERROR)
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_vpn_gateway.go]: this is AlibabaCloudSdkGoERROR

File: alicloud/service_alicloud_vpn_gateway.go [hunk 2]
```diff
@@ -145,26 +146,27 @@ func (s *VpnGatewayService) WaitForVpnGateway(id string, status Status, timeout
...
-			break
+			if NotFoundError(err) {
+				if status == Deleted {
+					return nil
+				}
+			} else {
+				return WrapError(err)
+			}
+		}
+		if object.CustomerGatewayId == id {
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_vpn_gateway.go]: please add status != Deleted

File: alicloud/service_alicloud_vpn_gateway.go [hunk 3]
```diff
@@ -37,24 +37,25 @@ func (s *VpnGatewayService) DescribeVpnGateway(id string) (v vpc.DescribeVpnGate
...
-	request.CustomerGatewayId = cgwId
+	request.CustomerGatewayId = id
 
 	raw, err := s.client.WithVpcClient(func(vpcClient *vpc.Client) (interface{}, error) {
 		return vpcClient.DescribeCustomerGateway(request)
 	})
 	if err != nil {
 		if IsExceptedErrors(err, []string{VpnForbidden, CgwNotFound}) {
-			return v, GetNotFoundErrorFromString(GetNotFoundMessage("VPN customer gateway", cgwId))
+			return v, WrapErrorf(err, NotFoundMsg, ProviderERROR)
```
Comment (YY-tomorrow) [inline alicloud/service_alicloud_vpn_gateway.go]: should use AlibabaCloudSdkGoERROR

File: alicloud/data_source_alicloud_vpn_gateways_test.go [hunk 4]
```diff
@@ -5,122 +5,14 @@ import (
...
-data "alicloud_vpn_gateways" "vpn_gateways" {
-	name_regex = "tf-testAcc-fake"
-}
-`
-
 func TestAccAlicloudVpnGatewaysDataSourceBasic(t *testing.T) {
 	rand := acctest.RandIntRange(1000, 9999)
+	preCheck := func() {
+		testAccPreCheck(t)
+		//testAccPreCheckWithAccountSiteType(t, IntlSite)
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_vpn_gateways_test.go]: Why remove the line 14

File: alicloud/resource_alicloud_vpn_customer_gateway_test.go [hunk 5]
```diff
@@ -97,179 +99,175 @@ func testSweepVPNCustomerGateways(region string) error {
...
+			if NotFoundError(err) {
+				continue
+			}
+			return err
+		}
+
+		if instance.CustomerGatewayId != "" {
+			return fmt.Errorf("VPN Customer Gateway %s still exist", instance.CustomerGatewayId)
+		}
+	}
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_vpn_customer_gateway_test.go]: I think the line 120-122 are redundant.

File: alicloud/resource_alicloud_vpn_customer_gateway_test.go [hunk 6]
```diff
@@ -97,179 +99,175 @@ func testSweepVPNCustomerGateways(region string) error {
...
+			continue
+		}
+
+		instance, err := vpnGatewayService.DescribeVpnCustomerGateway(rs.Primary.ID)
+
+		if err != nil {
+			if NotFoundError(err) {
+				continue
+			}
+			return err
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_vpn_customer_gateway_test.go]: wraperror(err)?

**Pull request merged**